LANDSCAPE/COMPARISONS/ARGOCD-FLUX/index.md

```diff
@@ -0,0 +1,275 @@
+
+
+## Argo CD
+
+Argo CD can also [integrate with a variety of Secret management tools](https://argo-cd.readthedocs.io/en/stable/operator-manual/secret-management/) to store these secrets in Git, safely. It leaves this choice up to the user. Some users have taken advantage of Argo CD’s plugin feature to create plugins, for example, the[ Argo CD Vault Plugin](https://github.com/IBM/argocd-vault-plugin). This “hands off” approach just treats secrets as another _Application_ in the cluster.
+
+
+# Webhook receivers
+
+Webhooks can be used to help with immediate reconciliation with traditional, event driven, workflows. This doesn’t take the place of the reconciliation loop, but rather acts as an “on demand” reconciliation if you don’t want to wait after making a commit to a repo.
+
+The two tools have similar approaches so we just highlight where to get more information.
+
+
+## FluxCD
+
+FluxCD includes support for incoming webhooks that make GitOps pull-based pipelines as fast and responsive as push-based pipelines. [Webhook receivers](https://fluxcd.io/docs/guides/webhook-receivers/) can notify the Flux controllers about changes in Git or Helm repositories, or other sources like Image Repositories.
+
+
+## Argo CD
+
+Argo CD also has a mechanism to receive [Webhooks](https://argo-cd.readthedocs.io/en/stable/operator-manual/webhook/) to shorten the time it takes to check for updates to a Git repository. Normally this is done every 3 minutes.
+
+
+# Alerting and notifications
+
+This is another area where both tools are very similar, so we will just focus on the specific implementations.
+
+
+## FluxCD
+
+FluxCD controllers emit events when resource statuses change, as new app versions are deployed, and when health checks pass or fail. [Alerting and outgoing notifications](https://fluxcd.io/docs/guides/notifications/) are configured with a Provider resource that connects outgoing webhooks, so Flux can provide status updates through different reporting channels such as Slack or embedded in Git Commit status checks.
+
+
+## Argo CD
+
+Argo CD uses a separate controller for alerting/notifications called [Argo CD Notifications](https://argocd-notifications.readthedocs.io/en/stable/). With Argo CD Notifications, a user can similarly set up providers and outgoing notifications to systems (like Slack or Pagerduty) altering of a specific event that occurred or if a sync failed or not. This project is[ in the process of being moved](https://github.com/argoproj-labs/argocd-notifications#argo-cd-notifications-is-now-part-of-argo-cd) into the main Argo CD codebase and will be part of the “core” Argo CD offering in future releases
+
+
+# Image updates automation
+
+Image updating is the process of notifying the GitOps controller that a new image has been created and pushed into a repository. This helps in expediting the process of rolling out new versions of applications.
```

Comment:
Is it worth calling out that this feature isn't directly tied into GitOps? I think it's great that it's being explored, but it's more related to CD.

Reply:
Hi @RedbackThomson Cc @christianh814

It is tied into GitOps.

• CI/CD is in [App Delivery TAG's charter](https://github.com/cncf/toc/blob/main/tags/app-delivery.md), and GitOps WG is within App Tag TAG.
• CD is essential in **strong** GitOps which has CD as a foundation, leading to CO (Continuous Operations).